packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTextInput.tsx

```diff
@@ -0,0 +1,276 @@
+import React, { FC } from 'react';
+import { CheckCircle, Warning } from '@phosphor-icons/react';
+
+import 'components/component_kit/new_designs/CWTextInput.scss';
+import { CWLabel } from '../cw_label';
+import { CWText } from '../cw_text';
+import type { ValidationStatus } from '../cw_validation_text';
+import { getClasses } from '../helpers';
+
+import { ComponentType } from '../types';
+
+type TextInputSize = 'small' | 'large';
+
+export type BaseTextInputProps = {
+  autoComplete?: string;
+  autoFocus?: boolean;
+  containerClassName?: string;
+  defaultValue?: string | number;
+  value?: string | number;
+  iconLeft?: JSX.Element;
+  iconLeftonClick?: () => void;
+  iconRight?: JSX.Element;
+  iconRightonClick?: () => void;
+  inputValidationFn?: (value: string) => [ValidationStatus, string] | [];
+  label?: string | React.ReactNode;
+  maxLength?: number;
+  isCompact?: boolean;
+  name?: string;
+  onInput?: (e) => void;
+  onenterkey?: (e) => void;
+  onClick?: (e) => void;
+  placeholder?: string;
+  tabIndex?: number;
+  instructionalMessage?: string;
+  manualStatusMessage?: string;
+  manualValidationStatus?: ValidationStatus;
+};
+
+type InputStyleProps = {
+  inputClassName?: string;
+  darkMode?: boolean;
+  disabled?: boolean;
+  size?: TextInputSize;
+  fullWidth?: boolean;
+  validationStatus?: ValidationStatus;
+  displayOnly?: boolean;
+};
+
+type InputInternalStyleProps = {
+  hasLeftIcon?: boolean;
+  hasRightIcon?: boolean;
+  isTyping?: boolean;
+};
+
+type TextInputProps = BaseTextInputProps &
+  InputStyleProps &
+  InputInternalStyleProps &
+  React.HTMLAttributes<HTMLDivElement>;
+
+type MessageRowProps = {
+  hasFeedback?: boolean;
+  label?: string | React.ReactNode;
+  statusMessage?: string;
+  validationStatus?: ValidationStatus;
+  instructionalMessage?: string;
+};
+
+export const MessageRow: FC<MessageRowProps> = ({
+  hasFeedback,
+  label,
+  statusMessage,
+  validationStatus,
+  instructionalMessage,
+}) => {
+  return (
+    <div className={getClasses({}, 'MessageRow')}>
+      {label && <CWLabel label={label} />}
+      {instructionalMessage && <CWLabel label={instructionalMessage} />}
+      {hasFeedback && (
+        <>
+          <div className={getClasses({ validationStatus }, 'icon')}>
+            {validationStatus === 'success' && <CheckCircle weight="fill" />}
+            {validationStatus === 'failure' && <Warning weight="fill" />}
+          </div>
+          <CWText
+            type="caption"
+            className={getClasses<{ status: ValidationStatus }>(
+              { status: validationStatus },
+              'feedback-message-text'
+            )}
+          >
+            {statusMessage}
+          </CWText>
+        </>
+      )}
+    </div>
+  );
+};
+
+export const useTextInputWithValidation = () => {
+  const [inputTimeout, setInputTimeout] = React.useState<
+    NodeJS.Timeout | undefined
+  >();
+  const [isTyping, setIsTyping] = React.useState<boolean>(false);
```

Comment:
do we need the information if user is typing? Figma indeed have typing as different state but it looks exactly the same as focus. The same as default vs filled - they are the same https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=493%3A4751&mode=dev